### DISCUSSION TITLE
Latency Metrics Logging

### DISCUSSION THREAD

**Discussion by ozheng-circle** [aws/aws-sdk-java-v2#5757] [Migration from v1 to v2] [answered]
SDK v1 could enable [latency metrics logging](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-logging.html#sdk-latency-logging), but I am having trouble finding the v2 counterpart? It seems like [this guide](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/logging-slf4j.html) is the only one that's relevant, and it doesn't have any information on latency logging.

**Comment by debora-ito** [comment, marked as answer]
In v2, you can generate latency metrics by using a MetricPublisher - 

```Java
        S3Client client = S3Client.builder()
            .region(Region.US_WEST_2)
            .overrideConfiguration(c->c.addMetricPublisher(LoggingMetricPublisher.create()))
            .build();
```

`LoggingMetricPublisher` is provided out-of-the-box, but there's also a way to publish metrics to CloudWatch.

For more info see [Use SDK Metrics](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/metrics.html) in our Dev Guide and check the examples in the [blog post](https://aws.amazon.com/blogs/developer/using-the-new-client-side-metrics-feature-in-the-aws-sdk-for-java-v2/).
